lib/hashie/extensions/coercion/active_model.rb

```diff
@@ -0,0 +1,85 @@
+require 'active_model/type'
+
+module Hashie
+  module Extensions
+    module Coercion
+      module ActiveModel
+        # Symbol is the one glaring type omission. Define it if it is not
+        # already defined.
+        begin
+          ::ActiveModel::Type.lookup(:symbol)
+        rescue ArgumentError
+          class Symbol < ::ActiveModel::Type::String
+            def type
+              :symbol
+            end
+
+            private
+
+            # The symbol type leverages in coercion logic in the String
+            # superclass and simply uses `to_sym` to covert to a symbol.
+            def cast_value(value)
+              super.to_sym
+            end
+          end
+          ::ActiveModel::Type.register(:symbol, Symbol)
+        end
+
+        # Array of symbols of the included ActiveModel types.
+        ACTIVE_MODEL_TYPES = ::ActiveModel::Type.registry
+                             .send(:registrations)
+                             .map { |r| r.send(:name) }
+                             .freeze
+
+        def build_coercion(type)
+          if type.is_a? Enumerable
+            if type.class <= ::Hash
+              type, key_type, value_type = type.class, *type.first
```

Comment:
This is a little confusing. I think this is the same and is a little clearer? I find the splat to still be confusing.

```ruby
key_type, value_type = *type.first
build_hash_coercion(type.class, key_type, value_type)
```